### PR TITLE
feat(console): dynamic sign in methods form

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Implement dynamic form for SignInMethods, including several cases when primary method changed.

You can check the [Figma Link](https://www.figma.com/file/iroUuWbeZ9fqcqXNwqYZai/Admin-Console-UI?node-id=589%3A17483) to see details.

## Linear Issue Reference

LOG-1736

## Testing

Local tested.

### CODE REVIEW

File: packages/console/src/components/Checkbox/index.tsx [hunk 1]
```diff
@@ -8,16 +8,16 @@ type Props = Omit<InputHTMLAttributes<HTMLInputElement>, 'type'> & {
   label?: ReactNode;
 };
 
-const Checkbox = ({ label, disabled, ...rest }: Props) => {
+const Checkbox = ({ label, disabled, ...rest }: Props, ref: LegacyRef<HTMLInputElement>) => {
```
Comment (demonzoo) [inline packages/console/src/components/Checkbox/index.tsx:11]: The type can just be `Ref<HTMLInputElement>`

File: packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx [hunk 2]
```diff
@@ -16,8 +16,40 @@ const signInMethods = Object.values(SignInMethodKey);
...
-  const { register, watch, control } = useFormContext<SignInExperienceForm>();
+  const { register, watch, control, setValue } = useFormContext<SignInExperienceForm>();
   const primaryMethod = watch('signInMethods.primary');
+  const enableSecondary = watch('signInMethods.enableSecondary');
+  const social = watch('signInMethods.social');
+
+  useEffect(() => {
+    if (primaryMethod) {
+      // Set secondary method to false.
+      setValue(`signInMethods.${primaryMethod}`, false);
```
Comment (demonzoo) [inline packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:27]: Does this comment match the code? Looks like you are setting `primaryMethod` to false

Comment (IceHe) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:27]: better?
```
// If one of the sign-in methods is primary, it should be unable to be set as secondary.
```

Comment (simeng-li) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:27]: non-identical guard

File: packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx [hunk 3]
```diff
@@ -16,8 +16,40 @@ const signInMethods = Object.values(SignInMethodKey);
...
+                    {t('sign_in_exp.sign_in_methods.methods_primary_tag')}
+                  </span>
+                )}
+              </>
+            }
+            disabled={primaryMethod === method}
+            {...register(`signInMethods.${method}`)}
+          />
+        </div>
+      )),
```
Comment (simeng-li) [inline packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:50]: nit: prefer avoid deep nested structor

```js
signInMethods.map((method) => {
  const label = (
             <>
                {t('sign_in_exp.sign_in_methods.methods', { context: method })}
                {primaryMethod === method && (
                  <span className={styles.primaryTag}>
                    {t('sign_in_exp.sign_in_methods.methods_primary_tag')}
                  </span>
                )}
              </>
    );
   
     return  <div key={method} className={styles.method}><Checkbox label={label} disabled={primaryMethod === method} {...register(`signInMethods.${method}`)}/></div>
})
```

[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:50]: or could even make the Lable as a in file sub-component

[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:50]: or could even make the Lable as a in file sub-component

**Review by simeng-li — APPROVED** [review]
overall LGTM

File: packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx]: updated

File: packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx]: updated

File: packages/console/src/components/Checkbox/index.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/components/Checkbox/index.tsx]: updated

**Pull request merged**